crates/core/src/pipeline.rs

```diff
@@ -61,21 +70,57 @@ impl Pipeline {
             ));
         }
 
+        let mut interval = tokio::time::interval(time::Duration::from_secs(10));
```

Comment:
we should discuss whether this should be customizable or not, i don't see why not